syntax = "proto3";
package core.protos.playbooks;

import "google/protobuf/wrappers.proto";
import "google/protobuf/struct.proto";


message Jenkins {
  message LastBuildDetails {
    google.protobuf.StringValue job_name = 1;
    google.protobuf.StringValue branch_name = 2; // Optional: Branch name for multibranch jobs
  }

  message GetAllBranches {
    google.protobuf.StringValue job_name = 1; // Multibranch job name
  }

  message RunJob {
    google.protobuf.StringValue job_name = 1;
    google.protobuf.StringValue parameters = 2;
    google.protobuf.Struct parameter_json = 3;
  }

  enum TaskType {
    UNKNOWN = 0;
    FETCH_LAST_BUILD_DETAILS = 1;
    RUN_JOB = 2;
    GET_ALL_BRANCHES = 3;
  }

  TaskType type = 1;
  oneof task {
    LastBuildDetails fetch_last_build_details = 2;
    RunJob run_job = 3;
    GetAllBranches get_all_branches = 4;
  }
}